File: apiwznm/WznmQQcoList.cpp
/**
	* \file WznmQQcoList.cpp
	* API code for table TblWznmQQcoList (implementation)
	* \copyright (C) 2016-2020 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 5 Dec 2020
  */
// IP header --- ABOVE

#include "WznmQQcoList.h"

using namespace std;
using namespace Sbecore;
using namespace Xmlio;

/******************************************************************************
 class WznmQQcoList
 ******************************************************************************/

WznmQQcoList::WznmQQcoList(
			const uint jnum
			, const string sref
			, const string Short
			, const string srefIxVBasetype
			, const string titIxVBasetype
			, const string stubQryRefWznmMQuery
			, const string stubRefWznmMTablecol
		) {
	this->jnum = jnum;
	this->sref = sref;
	this->Short = Short;
	this->srefIxVBasetype = srefIxVBasetype;
	this->titIxVBasetype = titIxVBasetype;
	this->stubQryRefWznmMQuery = stubQryRefWznmMQuery;
	this->stubRefWznmMTablecol = stubRefWznmMTablecol;
};

bool WznmQQcoList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "WznmQQcoList");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractStringUclc(docctx, basexpath, "sref", "srf", sref);
		extractStringUclc(docctx, basexpath, "Short", "sho", Short);
		extractStringUclc(docctx, basexpath, "srefIxVBasetype", "typ", srefIxVBasetype);
		extractStringUclc(docctx, basexpath, "titIxVBasetype", "typ2", titIxVBasetype);
		extractStringUclc(docctx, basexpath, "stubQryRefWznmMQuery", "qry", stubQryRefWznmMQuery);
		extractStringUclc(docctx, basexpath, "stubRefWznmMTablecol", "tco", stubRefWznmMTablecol);
	};

	return basefound;
};

/******************************************************************************
 class ListWznmQQcoList
 ******************************************************************************/

ListWznmQQcoList::ListWznmQQcoList() {
};

ListWznmQQcoList::ListWznmQQcoList(
			const ListWznmQQcoList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WznmQQcoList(*(src.nodes[i]));
};

ListWznmQQcoList& ListWznmQQcoList::operator=(
			const ListWznmQQcoList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WznmQQcoList(*(src.nodes[i]));

	return *this;
};

ListWznmQQcoList::~ListWznmQQcoList() {
	clear();
};

void ListWznmQQcoList::clear() {
	for (unsigned int i = 0; i < nodes.size(); i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListWznmQQcoList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	WznmQQcoList* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListWznmQQcoList");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "WznmQQcoList", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i = 0; i < ics.size(); i++) {
		rec = new WznmQQcoList(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/WznmQQcoList[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};

File: japiwznm/WznmQQcoList.java
/**
	* \file WznmQQcoList.h
	* Java API code for record of table TblWznmQQcoList
	* \copyright (C) 2018-2020 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 5 Dec 2020
  */
// IP header --- ABOVE

package apiwznm;

import org.w3c.dom.*;
import sbecore.*;

public class WznmQQcoList {
	
	public WznmQQcoList(
				int jnum
				, String sref
				, String Short
				, String srefIxVBasetype
				, String titIxVBasetype
				, String stubQryRefWznmMQuery
				, String stubRefWznmMTablecol
			) {
		this.jnum = jnum;
		this.sref = sref;
		this.Short = Short;
		this.srefIxVBasetype = srefIxVBasetype;
		this.titIxVBasetype = titIxVBasetype;
		this.stubQryRefWznmMQuery = stubQryRefWznmMQuery;
		this.stubRefWznmMTablecol = stubRefWznmMTablecol;
	};

	public int jnum;
	public String sref;
	public String Short;
	public String srefIxVBasetype;
	public String titIxVBasetype;
	public String stubQryRefWznmMQuery;
	public String stubRefWznmMTablecol;
	
	public boolean readXML(
				Document doc
				, String basexpath
				, boolean addbasetag
			) {
		if (addbasetag) basexpath = Xmlio.checkUclcXPaths(doc, basexpath, "WznmQQcoList");

		if (Xmlio.checkXPath(doc, basexpath)) {
			sref = Xmlio.extractStringUclc(doc, basexpath, "sref", "srf", null, 0);
			Short = Xmlio.extractStringUclc(doc, basexpath, "Short", "sho", null, 0);
			srefIxVBasetype = Xmlio.extractStringUclc(doc, basexpath, "srefIxVBasetype", "typ", null, 0);
			titIxVBasetype = Xmlio.extractStringUclc(doc, basexpath, "titIxVBasetype", "typ2", null, 0);
			stubQryRefWznmMQuery = Xmlio.extractStringUclc(doc, basexpath, "stubQryRefWznmMQuery", "qry", null, 0);
			stubRefWznmMTablecol = Xmlio.extractStringUclc(doc, basexpath, "stubRefWznmMTablecol", "tco", null, 0);
			
			return true;
		};

		return false;
	};
};

